pkg/cinderscheduler/statefulset.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ServiceCommand -
+	ServiceCommand = "/usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start"
+)
+
+// StatefulSet func
+func StatefulSet(
+	instance *cinderv1beta1.CinderScheduler,
+	configHash string,
+	labels map[string]string,
+) *appsv1.StatefulSet {
+	runAsUser := int64(0)
+
+	// TODO until we determine how to properly query for these
+	livenessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds:      5,
+		PeriodSeconds:       3,
+		InitialDelaySeconds: 3,
+	}
+	readinessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds:      5,
+		PeriodSeconds:       5,
+		InitialDelaySeconds: 5,
+	}
+
+	startupProbe := &corev1.Probe{
+		TimeoutSeconds:      5,
+		FailureThreshold:    12,
+		PeriodSeconds:       5,
+		InitialDelaySeconds: 5,
+	}
+
+	args := []string{"-c"}
+	if instance.Spec.Debug.Service {
+		args = append(args, common.DebugCommand)
+		livenessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
+
+		readinessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
+	} else {
+		args = append(args, ServiceCommand)
+		livenessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/usr/local/bin/container-scripts/healthcheck.sh",
+				"cinder-scheduler",
+			},
+		}
+
+		readinessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/usr/local/bin/container-scripts/healthcheck.sh",
+				"cinder-scheduler",
+			},
+		}
+
+		startupProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/usr/local/bin/container-scripts/healthcheck.sh",
+				"cinder-scheduler",
+			},
+		}
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_FILE"] = env.SetValue(KollaConfig)
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+
+	statefulset := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instance.Name,
+			Namespace: instance.Namespace,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Replicas: &instance.Spec.Replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: cinder.ServiceAccount,
+					Containers: []corev1.Container{
+						{
+							Name: cinder.ServiceName + "-scheduler",
+							Command: []string{
+								"/bin/bash",
+							},
+							Args:  args,
+							Image: instance.Spec.ContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								RunAsUser: &runAsUser,
+							},
+							Env:          env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts: cinder.GetVolumeMounts(),
+							// TODO?
```

Comment:
yes I think we should have this per default for all services, also the affinity one

Reply:
I'll add the `Resources` config, but I think the affinity stuff is already there for all sub-services (following the pattern currently found in placement operator)